packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -0,0 +1,189 @@
+
+		for (const mapping of this.subcommands) {
+			if (!(mapping instanceof MessageSubcommandMappings)) continue;
+			defaultCommmand = mapping.subcommands.find((s) => s.default === true) ?? null;
+			const subcommand = mapping.subcommands.find(({ name }) => name === value.value);
+			if (subcommand) return this.#handleMessageRun(message, args, context, subcommand);
+		}
+
+		// No subcommand matched, let's restore and try to run default, if any:
+		args.restore();
+		if (defaultCommmand) return this.#handleMessageRun(message, args, context, defaultCommmand);
+
+		// No match and no subcommand, return an err:
+		throw new UserError({ identifier: Identifiers.MessageSubcommandNoMatch, context });
+	}
+
+	public chatInputRun(interaction: ChatInputCommand.Interaction, context: ChatInputCommand.RunContext): Awaitable<unknown> {
+		const subcommandName = interaction.options.getSubcommand(false);
+		const subcommandGroupName = interaction.options.getSubcommandGroup(false);
+
+		if (subcommandName && !subcommandGroupName) {
+			for (const mapping of this.subcommands) {
+				if (!(mapping instanceof ChatInputSubcommandMappings)) continue;
+
+				const subcommand = mapping.subcommands.find(({ name }) => name === subcommandName);
+				if (subcommand) return this.#handleInteractionRun(interaction, context, subcommand);
+			}
+		}
+
+		if (subcommandGroupName) {
+			for (const mapping of this.subcommands) {
+				if (!(mapping instanceof ChatInputSubcommandGroupMappings)) continue;
+				if (mapping.groupName !== subcommandGroupName) continue;
+
+				const subcommand = mapping.subcommands.find(({ name }) => name === subcommandName);
+				if (subcommand) return this.#handleInteractionRun(interaction, context, subcommand);
+			}
+		}
+
+		// No match and no subcommand, return an err:
+		throw new UserError({ identifier: Identifiers.ChatInputSubcommandNoMatch, context });
+	}
+
+	async #handleInteractionRun(
+		interaction: ChatInputCommand.Interaction,
+		context: ChatInputCommand.RunContext,
+		subcommand: ChatInputSubcommandMappingValue
+	) {
+		const payload: ChatInputSubcommandAcceptedPayload = { command: this, context, interaction };
+		const result = await fromAsync(async () => {
+			interaction.client.emit(Events.ChatInputSubcommandRun, interaction, subcommand, payload);
+			let result: unknown;
+			subcommand.type ??= 'method';
+
+			if (subcommand.type === 'command') {
+				const parsedCommandName = subcommand.to && typeof subcommand.to === 'string' ? subcommand.to : subcommand.name;
+				const command = this.container.stores.get('commands').get(parsedCommandName);
+				if (!command?.chatInputRun) throw new UserError({ identifier: Identifiers.SubcommandNotFound, context: { ...payload } });
+
+				// Run global preconditions:
+				const globalResult = await this.container.stores
+					.get('preconditions')
+					.chatInputRun(interaction, command as ChatInputCommand, context as any);
+				if (!globalResult.success) {
```

Comment:
```suggestion
					.chatInputRun(interaction, command as ChatInputCommand, context as any);

				if (!globalResult.success) {
```

Reply:
this diff looks wonky

newline between if

Reply:
Hahaha same